vme/src/essential.h

```diff
@@ -118,6 +118,7 @@ using ubit1 = bool;      /* Boolean */
 
 #define FREE(p)                                                                                                                            \
     {                                                                                                                                      \
+        /* Make sure FREE isn't being used to free class items created with EMPLACE */                                                     \
```

Comment:
I was wondering if you made a unit_test for this code? It looks complicated, not sure I'd know how to test it if I had to :-)

Reply:
Don't really have to, its a compile time check - it won't compile if you pass in a class pointer :-)

But hopefully this template voodoo will go away as things get cleaned up